### PR TITLE
Custom games - game search API

### PR DESCRIPTION
**Pull request by SlyCaptainFlint**
Adding graphql query for searching for games by a search term (returns max of 5 results for now). This will also add games to our database if they are not already there

### CODE REVIEW

File: src/AllMyGamesAPI/AllMyGamesAPI.csproj [hunk 1]
```diff
@@ -62,4 +61,8 @@
     <Folder Include="Services\GameLibrary\" />
     <Folder Include="Services\Following\" />
   </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\..\IgdbDotNet\src\IgdbDotNet\IgdbDotNet.csproj" />
```
Comment (MChartier) [inline src/AllMyGamesAPI/AllMyGamesAPI.csproj:66]: Consume the latest package from NuGet.

File: src/AllMyGamesAPI/GraphQL/AllMyGamesQuery.cs [hunk 2]
```diff
@@ -17,7 +21,10 @@ namespace AllMyGamesAPI.GraphQL
 {
     public class AllMyGamesQuery : ObjectGraphType
     {
-        public AllMyGamesQuery(IServiceScopeFactory<ApplicationDbContext> dbCtxFactory)
+        public AllMyGamesQuery(
+            IServiceScopeFactory<ApplicationDbContext> dbCtxFactory, 
+            IProfileRepository profileRepository,
+            IIgdbGameRepository igdbGameRepository)
```
Comment (MChartier) [inline src/AllMyGamesAPI/GraphQL/AllMyGamesQuery.cs:27]: I just wiped out all of these repository classes and interfaces, which you'll see in a merge conflict.

You can restore `IgdbGameRepository`, or alternatively pull the method you need from it into this class instead.

The method on `IProfileRepository` has been moved to the new static class `AllMyGamesGraphQLUtils`.

File: src/AllMyGamesAPI/GraphQL/AllMyGamesQuery.cs [hunk 3]
```diff
@@ -104,6 +111,44 @@ public AllMyGamesQuery(IServiceScopeFactory<ApplicationDbContext> dbCtxFactory)
...
+                "searchGame",
+                arguments: new QueryArguments(new QueryArgument<NonNullGraphType<SearchGameQueryInputType>> { Name = "input" }),
+                resolve: async context =>
+                {
+                    const int RESULTS_TO_DISPLAY = 5;
+                    var input = context.GetArgument<SearchGameQueryInput>("input");
+
+                    if (string.IsNullOrEmpty(input.SearchTerm))
+                    {
+                        return null;
```
Comment (MChartier) [inline src/AllMyGamesAPI/GraphQL/AllMyGamesQuery.cs:125]: Please add something like this above the lines where you are returning null.

`context.Errors.Add(new ExecutionError("Reason for failing."));`

Not quite a fully-realized error handling strategy (yet) but I think this at least causes a non-success response to be returned which can be handled by the client.

**Comment by MChartier** [comment]
I added a commit to this branch to fix some of the things that were broken in the merge as a result of my changes from yesterday.

File: src/AllMyGamesAPI/GraphQL/AllMyGamesQuery.cs (hunk 2 above)
Comment (MChartier) [reply src/AllMyGamesAPI/GraphQL/AllMyGamesQuery.cs:26]: I added a commit to fix this.

File: src/AllMyGamesAPI/AllMyGamesAPI.csproj (hunk 1 above)
Comment (MChartier) [reply src/AllMyGamesAPI/AllMyGamesAPI.csproj]: I added a commit to fix this.

**Comment by MChartier** [comment]
Just tried this out along with the in-development client change. Leaving a few observations here:
1. Search results include DLC and "editions" of games. E.g., a search for "Mario Kart 8" yields not only the two versions of the game for Wii U and Switch, but also two DLC packs and "Mario Kart 8 Limited Edition". Not sure we should include DLC packs, and definitely shouldn't include things like the Limited Edition here.
2. Search bar - interactive clickable portion doesn't take up full visible width.
3. When I dismiss the modal dialog and then open it again, the search query has been cleared but the results are still there. Should reset both or neither. (Interestingly, though, if I click "Find Games" again in this case it performs the same search - evidently the search term is persisted in the state of the component but just isn't showing up in the UI!)

**Comment by SlyCaptainFlint** [comment]
Comments 2-4 are about the client, so I'm going to handle them as part of that PR. I'm not really sure how we filter out undesirable content you call out in 1. Do you suggest we exclude things based on keywords?

File: src/AllMyGamesAPI/libman.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (MChartier) [inline src/AllMyGamesAPI/libman.json:1]: What is this file?

Comment (SlyCaptainFlint) [reply src/AllMyGamesAPI/libman.json]: wow, no idea. deleting it didn't seem to hurt anything, though

**Pull request merged**